automation_tools/satellite6/upgrade/capsule.py

```diff
@@ -0,0 +1,161 @@
+            sys.exit(1)
+        cap_image = os.environ.get('RHEV_CAP_IMAGE')
+        cap_hosts = os.environ.get('RHEV_CAP_HOST')
+        cap_instance = 'upgrade_capsule_auto_{0}'.format(os_version)
+        # Deleting Capsule instance if already exists
+        execute(delete_rhevm_instance, cap_instance)
+        print('Turning on Capsule Instance ....')
+        execute(create_rhevm_instance, cap_instance, cap_image)
+        # Restarting the services on capsule
+        execute(lambda: run('katello-service restart'), host=cap_hosts)
+    # Set capsule hostname in fabric environment
+    env['capsule_hosts'] = cap_hosts
+    # Copy ssh key from satellie to capsule
+    copy_ssh_key(sat_host, cap_hosts)
+    if os.environ.get('CAPSULE_URL') is not None:
+        execute(sync_capsule_repos_to_upgrade, cap_hosts, host=sat_host)
+
+
+def satellite6_capsule_upgrade(cap_host):
+    """Upgrades capsule from existing version to latest version.
+
+    :param string cap_host: Capsule hostname onto which the capsule upgrade
+    will run
+
+    The following environment variables affect this command:
+
+    CAPSULE_URL
+        Optional, defaults to available capsule version in CDN.
+        URL for capsule of latest compose to upgrade.
+    FROM_VERSION
+        Capsule current version, to disable repos while upgrading.
+        e.g '6.1','6.0'
+    TO_VERSION
+        Capsule version to upgrade to and enable repos while upgrading.
+        e.g '6.1','6.2'
+
+    """
+    sat_host = env.get('satellite_host')
+    from_version = os.environ.get('FROM_VERSION')
+    to_version = os.environ.get('TO_VERSION')
+    # Setup firewall rules on capsule node
+    setup_capsule_firewall()
+    # Setting Capsule61 Repos
+    major_ver = distro_info()[1]
+    # Re-register Capsule for 6.2
+    # AS per host unification feature: if there is a host registered where the
+    # Host and Content Host are in different organizations (e.g. host not in
+    # org, and content host in one), the content host will be unregistered as
+    # part of the upgrade process.
+    if to_version == '6.2':
+        ak_name = os.environ.get('CAPSULE_SUBSCRIPTION').split(',')[2].strip(
+            ) if os.environ.get('CAPSULE_SUBSCRIPTION') else os.environ.get(
+            'RHEV_CAPSULE_AK')
+        run('subscription-manager register --org="Default_Organization" '
+            '--activationkey={0} --force'.format(ak_name))
+    # if CDN Upgrade enable cdn repo
+    if os.environ.get('CAPSULE_URL') is None:
+        enable_repos('rhel-{0}-server-satellite-capsule-{1}-rpms'.format(
+            major_ver, to_version))
+    disable_repos('rhel-{0}-server-satellite-capsule-{1}-rpms'.format(
+        major_ver, from_version))
+    if from_version == '6.1' and major_ver == '6':
+        enable_repos('rhel-server-rhscl-{0}-rpms'.format(major_ver))
+    if from_version == '6.0':
+        # Stop katello services, except mongod
+        run('for i in qpidd pulp_workers pulp_celerybeat '
```

Comment:
We cannot run katello-service stop for 6.0 for rest we can. The command itself is not present there in 6.0 capsule